doc/tutorial/using_gpu.txt

```diff
@@ -532,23 +192,61 @@ What Can be Accelerated on the GPU
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
 The performance characteristics will of course vary from device to
-device, and also as we refine our implementation.
-
-This backend supports all regular theano data types (float32, float64,
-int, ...) however GPU support varies and some units can't deal with
+device, and also as we refine our implementation:
+
+* In general, matrix multiplication, convolution, and large element-wise
+  operations can be accelerated a lot (5-50x) when arguments are large enough
+  to keep 30 processors busy.
+* Indexing, dimension-shuffling and constant-time reshaping will be equally fast
+  on GPU as on CPU.
+* Summation over rows/columns of tensors can be a little slower on the
+  GPU than on the CPU.
+* Copying of large quantities of data to and from a device is relatively slow,
+  and often cancels most of the advantage of one or two accelerated functions
+  on that data. Getting GPU performance largely hinges on making data transfer
+  to the device pay off.
+
+The backend supports all regular theano data types (float32, float64,
+int, ...), however GPU support varies and some units can't deal with
 double (float64) or small (less than 32 bits like int16) data types.
 You will get an error at compile time or runtime if this is the case.
 
-By default all inputs will get transferred to GPU.  You can prevent an
+By default all inputs will get transferred to GPU. You can prevent an
 input from getting transferred by setting its tag.target attribute to
 'cpu'.
 
 Complex support is untested and most likely completely broken.
 
-In general, large operations like matrix multiplication, or
-element-wise operations with large inputs, will be significatly
-faster.
+Tips for Improving Performance on GPU
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
+* Consider adding ``floatX=float32`` (or the type you are using) to your
+  ``.theanorc`` file if you plan to do a lot of GPU work.
+* The GPU backend supports *float64* variables, but they are still slower
+  to compute than *float32*. The more *float32*, the better GPU performance
+  you will get. 
+* Prefer constructors like ``matrix``, ``vector`` and ``scalar`` to
+  ``dmatrix``, ``dvector`` and ``dscalar`` because the former will give
+  you *float32* variables and ignore the type given to ``floatX``.
```

Comment:
This is said in a confusing way. `matrix`, etc., *will follow* `floatX`, so they will be float32 when `floatX==float32`.
`dmatrix`, etc. *will ignore* `floatX` and use *float64* all the time.